lib/vnet/daemon/client_darwin.go

```diff
@@ -0,0 +1,197 @@
+// #cgo LDFLAGS: -framework Foundation -framework ServiceManagement
+// #include "client_darwin.h"
+import "C"
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os/exec"
+	"time"
+	"unsafe"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport"
+	logutils "github.com/gravitational/teleport/lib/utils/log"
+)
+
+var log = logutils.NewPackageLogger(teleport.ComponentKey, "vnet:daemon")
+
+// Taken from manual for codesign.
+const codesignExitCodeVerificationFailed = 1
+
+func IsSigned(ctx context.Context) (bool, error) {
+	var result C.BundlePathResult
+	defer func() {
+		C.free(unsafe.Pointer(result.bundlePath))
+	}()
+	C.BundlePath(&result)
+
+	bundlePath := C.GoString(result.bundlePath)
+	if bundlePath == "" {
+		return false, nil
+	}
+
+	if err := exec.CommandContext(ctx, "codesign", "--verify", bundlePath).Run(); err != nil {
+		var exitError *exec.ExitError
+		if errors.As(err, &exitError) {
+			log.DebugContext(ctx, "codesign --verify returned with non-zero exit code",
+				"exit_code", exitError.ExitCode(), "bundle_path", bundlePath)
+
+			if exitError.ExitCode() == codesignExitCodeVerificationFailed {
+				return false, nil
+			}
+
+			// Returning a custom error instead of wrapping err because err is just a cryptic "exit status 2".
+			return false, trace.Errorf("failed to check the signature of tsh, codesign returned with exit code %d", exitError.ExitCode())
+		}
+
+		return false, trace.Wrap(err, "failed to check the signature of tsh")
+	}
+
+	return true, nil
+}
+
+// RegisterAndCall attempts to register the daemon as a login item, waits for the user to enable it
+// and then starts it by sending a message through XPC.
+func RegisterAndCall(ctx context.Context, socketPath, ipv6Prefix, dnsAddr string) error {
+	initialStatus := daemonStatus()
+	// If the status is equal to "requires approval" before RegisterAndCall called register, it means
+	// that it's not the first time the user tries to start the daemon. In that case, macOS is not
+	// going to show the notification about a new login item. Instead, we just open the login items
+	// ourselves to direct the user towards enabling the login item.
+	if initialStatus == serviceStatusRequiresApproval {
+		C.OpenSystemSettingsLoginItems()
+	}
+
+	status, err := register(ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if status != serviceStatusEnabled {
+		if err := waitForEnablement(ctx); err != nil {
+			return trace.Wrap(err, "waiting for the login item to get enabled")
+		}
+	}
+
+	return trace.NotImplemented("RegisterAndCall is not fully implemented yet")
+}
+
+func register(ctx context.Context) (serviceStatus, error) {
+	var result C.RegisterDaemonResult
+
+	C.RegisterDaemon(&result)
+	defer func() {
+		C.free(unsafe.Pointer(result.error_description))
+	}()
+
+	if !result.ok {
+		status := serviceStatus(result.service_status)
+		// Docs for registerAndReturnError [1] don't seem to cover a situation in which the user adds
```

Comment:
nit: You have to "anchor" the link to something:

```suggestion
		// Docs for [registerAndReturnError][1] don't seem to cover a situation in which the user adds
```